src/mokernels/independent.jl

```diff
@@ -23,18 +23,44 @@ struct IndependentMOKernel{Tkernel<:Kernel} <: MOKernel
     kernel::Tkernel
 end
 
+# kernel function should be symmetric
+# would really like (κ::IndependentMOKernel)((x, px)::Tuple{T,Int}, (y, py)::Tuple{T,Int}) where T, but seems to cause autodiff problems
 function (κ::IndependentMOKernel)((x, px)::Tuple{Any,Int}, (y, py)::Tuple{Any,Int})
-    if px == py
-        return κ.kernel(x, y)
-    else
-        return 0.0
-    end
+    return κ.kernel(x, y) * (px == py)
+end
+
+function _kronkernelmatrix(Ktmp, B, ::MOInputIsotopicByFeatures)
```

Comment:
It's starting to get a bit confusing with _kronkernelmatrix, _kroneckerkernelmatrix, kernelkronmat... how about at least clarifying in the method name that this is multioutput-specific? e.g.
```suggestion
function _mokernelmatrix_kron(::MOInputIsotopicByFeatures, Kfeatures, Koutputs)
```
(moving the argument on which we actually dispatch to the front)

Reply:
I'm not quite happy with this being a different pattern from kernelkronmat when they should otherwise be the same, `kron` vs `Kronecker` being the only difference. Could it be a single method that returns either dense kron() or lazy Kronecker? If that would lead to type instability issues, could dispatch on a Val{} maybe? lazy=Val{true} vs lazy=Val{false} (it's just an internal function anyways)